test/models/session/PensionCYAMergedWithPriorDataSpec.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.session
+
+import builders.PensionsUserDataBuilder._
+import models.mongo.{PensionsCYAModel, PensionsUserData}
+import models.pension.AllPensionsData
+import models.pension.charges._
+import models.pension.reliefs.PaymentsIntoPensionsViewModel
+import models.pension.statebenefits.{IncomeFromPensionsViewModel, StateBenefitViewModel}
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AnyWordSpecLike
+import utils.PensionDataStubs._
+import java.util.UUID
+
+class PensionCYAMergedWithPriorDataSpec extends AnyWordSpecLike with TableDrivenPropertyChecks {
+
+  "mergeSessionAndPriorData" should {
+    val emptyModel            = PensionsCYAModel.emptyModels
+    val sessionWithEmptyModel = aPensionsUserData.copy(pensions = emptyModel)
+    val expectedFullSession   = AllPensionsData.generateCyaFromPrior(fullPensionsModel)
+    val sessionFullModel      = sessionWithEmptyModel.copy(pensions = expectedFullSession)
+    val sessionNoPaymentIntoPension =
+      sessionWithEmptyModel.copy(pensions = expectedFullSession.copy(paymentsIntoPension = PaymentsIntoPensionsViewModel()))
+
+    val cases = Table(
+      ("sessionData", "priorData", "expectedResult"),
+      (None, None, PensionCYAMergedWithPriorData(emptyModel, false)),
+      (Some(sessionWithEmptyModel), None, PensionCYAMergedWithPriorData(emptyModel, false)),
+      (None, Some(fullPensionsModel), PensionCYAMergedWithPriorData(expectedFullSession, true)),
+      (Some(sessionWithEmptyModel), Some(fullPensionsModel), PensionCYAMergedWithPriorData(expectedFullSession, true)),
+      (Some(sessionFullModel), Some(fullPensionsModel), PensionCYAMergedWithPriorData(expectedFullSession, false)),
+      (Some(sessionNoPaymentIntoPension), Some(fullPensionsModel), PensionCYAMergedWithPriorData(expectedFullSession, true))
+    )
+
+    def assertMergeCases(sessionData: Option[PensionsUserData], priorData: Option[AllPensionsData], expectedResult: PensionCYAMergedWithPriorData) =
+      assert(PensionCYAMergedWithPriorData.mergeSessionAndPriorData(sessionData, priorData) === expectedResult)
+
+    "merge session with prior data" in forAll(cases)(assertMergeCases)
+
+    // let's test if every journey with at least one field filled will stop prior data from being populated
+    // (user is in the middle of the change, don't override their data)
```

Comment:
Do we need this comment?

Reply:
I think so, the tests are not immediately obvious what they do - I used antipattern numeric pension1, pension2... to not pollute code with long names.